### PR TITLE
Move http request code for package metadata to master.py

### PR DESCRIPTION
**Pull request by dralley**
It feels like it makes more sense as a utility function there, beside all the other API calls.

### CODE REVIEW

File: src/bandersnatch/package.py [hunk 1]
```diff
@@ -142,17 +141,13 @@ async def sync(self, stop_on_error: bool = False, attempts: int = 3) -> None:
...
-                        if e.status == 404:
-                            logger.info(f"{self.name} no longer exists on PyPI")
-                            return None
-                        raise
+
+                    metadata = await self.mirror.master.get_package_metadata(
+                        self.name, serial=self.serial
+                    )
+                    if not metadata:
+                        return None
```
Comment (dralley) [inline src/bandersnatch/package.py:149]: Would a custom error type be better?  e.g. have the ```get_package_metadata``` function ```raise PackageNotFound(f"{self.name} no longer exists on PyPI"``` and then catch and log and return as before?

**Comment by cooperlees** [comment]
Travis reliability has gone downhill lately :|

**Comment by dralley** [comment]
It has, we've had issues with it as well :/

**Comment by dralley** [comment]
Looks like it's good now though

**Pull request merged**